Heroku deployment

tingamapuro04 opened the pull request:
The presentation video and the link to the live demo are in the readme

Splash screen

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
Home page (catagories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
Testing requirements
Create unit and integration tests for all the most important components of your RoR application.
Technical requirements
You should use Postgres as your database.
You should use devise for authentication.
You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
You can use a view template engine of your choice (.erb, .slim, .haml).
The project should be deployed and accessible online.
Your database schema should reflect the following structure:

tingamapuro04 committed:
Add live demo

tingamapuro04 committed:
Add presentation video

FrederickMih commented on README.md:
```diff
@@ -12,6 +12,11 @@ An app where a user is able to select items based on a category and also be able
+[Purchase-App]('https://naivas-purchase.herokuapp.com')
+
+## Video Presentation
+[Presentation]('https://www.loom.com/share/1db585d82cb54eefb652e5e4e0d35908')
```
- [ ] These are not clickable links. Let's kindly make sure that clicking on each link redirects us to the live version and video presentation.

FrederickMih commented on README.md:
```diff
@@ -12,6 +12,11 @@ An app where a user is able to select items based on a category and also be able
+[Presentation]('https://www.loom.com/share/1db585d82cb54eefb652e5e4e0d35908')
 
 
 ## Getting Started
```
- [ ] Also, remove these instructions(**This is an example of how you may give instructions on setting up your project locally. Modify this file to match your project, remove sections that don't apply. For example: delete the testing section if the currect project doesn't require testing.**) in the Getting Started section of the README file. They are directives from the template. You were not supposed to allow it in your README.
- [ ] kindly provide instructions on how to set up the project locally in the Getting Started section. Please include all commands required.

FrederickMih replied:
- [ ] The License link MIT is broken. Please fix :wrench: this issue, this way, our README will become professional.

FrederickMih reviewed (changes requested):
Hi @tingamapuro04, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!


![Screenshot from 2022-10-20 09-38-51](https://user-images.githubusercontent.com/63087491/198282606-41ba323f-a1e6-4c1e-8d20-44e30fd9f375.png)


## Highlights :green_circle: 

- PR has a summary :+1: 
- No Linter error :heavy_check_mark: 


## Suggested changes :red_circle: 

- **NB** You made fewer changes in your files. It becomes very difficult to leave a comment below a line of code. This is the reason I'm leaving some comments here. Let's kindly fix :wrench: them.

- [ ] I think the title(**Full project #9**) of this PR is not descriptive. Remember that this is a project which may be added to your Portfolio website. Let's make the PR more descriptive by editing the title to be descriptive.
- [ ] Based on the requirements of this project, you are supposed to highlight the **piece of code** you found interesting while developing your app. You only mentioned being proud of the design. You were required to follow that design. Kindly Highlight to us the piece of code you are proud of making.
- [ ] We are supposed to follow the provided design that was given in the requirements. Let's make sure that all calls to action buttons should be positioned below as seen in [this design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)

- [ ] Based on the requirements of this project, for each category, the user can see their name :+1: , icon :+1:, and the total amount of the transactions that belong to that category. You didn't show up the total amount of transactions. Instead, you did a count of the created transactions. Please kindly fix :wrench: this. 
- [ ] When I decided to create a group without an image, I got the error on the screenshot below:

![Screenshot from 2022-10-27 14-14-02](https://user-images.githubusercontent.com/63087491/198294528-37f57df2-01f9-4a9a-bebe-eff1da98c6ad.png)

- [ ] When I fill out the form to create a new transaction without the image of field, I got the same error on the screenshot above :point_up: . Let's kindly fix :wrench: it.
- [ ] After getting those errors above, your app was no longer working again. I couldn't register, sign in, create a group or add a purchase. Kindly make sure that your app is working perfectly before submitting it for another review.

_Check the comments under the review._


## Optional Suggested 

- I think it'll be nice to also include the date of creation of each group(category). This will improve the user experience.

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

tingamapuro04 committed:
Update html

tingamapuro04 committed:
Add style for group button

tingamapuro04 committed:
Validate that an image is chosen for either pyrchase or group

tingamapuro04 committed:
Correct Linter Errors

tingamapuro04 committed:
Update readme file

tingamapuro04 committed:
Update readme.md

tingamapuro04 committed:
Code refactor

tingamapuro04 committed:
Merge pull request #11 from tingamapuro04/review

Review

tingamapuro04 commented:
I have implemented the suggested changes such as
- One cannot create a category without an icon
- In every group, I display the total amount related to that category
- Added a descriptive pr title
- The links in the readme are now clickable

elisha2kyakpo1 reviewed (approved):
Hi 👋 @tingamapuro04, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You did well to make the changes requested ✔️✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tingamapuro04 committed:
remove confirmable from devise

tingamapuro04 committed:
Make page responsive

tingamapuro04 committed:
Merge pull request #12 from tingamapuro04/review

Review

tingamapuro04 commented:
All project requirements implemented
